test/e2e/e2e_test.go

```diff
@@ -25,101 +26,128 @@ import (
-			ExpectWithOffset(1, err).NotTo(HaveOccurred())
+			Expect(err).NotTo(HaveOccurred())
 
-			By("deploying the controller-manager")
-			//nolint:gosec
+			By("deploying the operator")
 			cmd = exec.Command("make", "deploy", fmt.Sprintf("IMG=%s", projectimage))
 			_, err = utils.Run(cmd)
-			ExpectWithOffset(1, err).NotTo(HaveOccurred())
-
-			By("validating that the controller-manager pod is running as expected")
-			verifyControllerUp := func() error {
-				// Get pod name
-				//nolint:gosec
-				cmd = exec.Command("kubectl", "get",
-					"pods", "-l", "control-plane=controller-manager",
-					"-o", "go-template={{ range .items }}"+
-						"{{ if not .metadata.deletionTimestamp }}"+
-						"{{ .metadata.name }}"+
-						"{{ \"\\n\" }}{{ end }}{{ end }}",
-					"-n", namespace,
-				)
-
-				podOutput, err := utils.Run(cmd)
-				ExpectWithOffset(2, err).NotTo(HaveOccurred())
-				podNames := utils.GetNonEmptyLines(string(podOutput))
-				if len(podNames) != 1 {
-					return fmt.Errorf("expect 1 controller pods running, but got %d", len(podNames))
+			Expect(err).NotTo(HaveOccurred())
+
+			By("validating that the operator pod is running")
+			Eventually(func() error {
+				pods, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{
+					LabelSelector: "control-plane=ceph-csi-op-controller-manager",
+				})
+				if err != nil {
+					return err
+				}
+				if len(pods.Items) != 1 {
+					return fmt.Errorf("expected 1 operator pod, but got %d", len(pods.Items))
 				}
-				controllerPodName = podNames[0]
-				ExpectWithOffset(2, controllerPodName).Should(ContainSubstring("controller-manager"))
-
-				// Validate pod status
-				//nolint:gosec
-				cmd = exec.Command("kubectl", "get", "pods", controllerPodName, "-o", "jsonpath={.status.phase}", "-n", namespace)
-				status, err := utils.Run(cmd)
-				ExpectWithOffset(2, err).NotTo(HaveOccurred())
-				if string(status) != "Running" {
-					return fmt.Errorf("controller pod in %s status", status)
+				if pods.Items[0].Status.Phase != corev1.PodRunning {
+					return fmt.Errorf("operator pod is in %s status", pods.Items[0].Status.Phase)
 				}
 				return nil
+			}, time.Minute, time.Second).Should(Succeed())
+		})
+	})
+
+	Context("Driver CR", func() {
+		It("should create a RBD Driver CR and verify resources", func() {
+			driver := &csiv1alpha1.Driver{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "csi.ceph.io/v1alpha1",
+					Kind:       "Driver",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test.rbd.csi.ceph.com",
+					Namespace: namespace,
+				},
+				Spec: csiv1alpha1.DriverSpec{
+					AttachRequired:   ptr.To(true),
+					NodePlugin:       &csiv1alpha1.NodePluginSpec{},
+					ControllerPlugin: &csiv1alpha1.ControllerPluginSpec{},
+				},
 			}
-			EventuallyWithOffset(1, verifyControllerUp, time.Minute, time.Second).Should(Succeed())
 
+			By("creating a Driver CR")
+			Expect(k8sClient.Create(context.TODO(), driver)).To(Succeed())
```

Comment:
These are not unit tests these are e2e tests and should be code from a kubectl perspective 
please use Command.Exec and a fixed file to read from